src/server/index.js

```diff
@@ -120,6 +120,23 @@ router.post('/boards', function(req, res) {
   }
 });
 
+router.delete('/boards/:id', function(req, res) {
+  let username = jwtVerify(req.headers.token);
+  let boardId = req.params.id;
+
+  if (!username) {
+    res.status(STATUS_FORBIDDEN).json({message: 'Please login first!'});
+  } else {
+    boards.deleteboardId(username, boardId, function(result) {
+      if (result === 'error' || !result || result.result.n === 0) {
```

Comment:
still if `result.result.n === 0` that means we could not find a board with the specified id so it's not an internal software error and should respond with 404 not found.